internal/variants/variants.go

```diff
@@ -71,8 +81,29 @@ func (q *Query) ValidateInput() error {
 	if q.Allele == "" {
 		return errors.New("missing allele")
 	}
-	if q.Coord == nil {
-		return errors.New("missing coordinate")
+	if err := q.validateCoordinates(); err != nil {
+		return fmt.Errorf("validating coordinates: %v", err)
+	}
+	return nil
+}
+
+func (q *Query) validateCoordinates() error {
+	var precisePosition, imprecisePosition bool
+	if q.Start != nil && (q.End != nil || q.Allele != "") {
+		precisePosition = true
```

Comment:
By not specifying the `q.End` and only the `q.Allele` that is the definition of an *imprecise* position, as it is open-ended on one side.